tests/unit/machine/Pipeline_unittest.cc

```diff
@@ -0,0 +1,101 @@
+#include "features/MockFeatures.h"
+#include "labels/MockLabels.h"
+#include "machine/MockMachine.h"
+#include "transformer/MockTransformer.h"
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <shogun/base/some.h>
+#include <shogun/machine/Pipeline.h>
+
+using namespace shogun;
+using ::testing::Mock;
+using ::testing::NiceMock;
+using ::testing::Return;
+using ::testing::_;
+using ::testing::InSequence;
+
+class PipelineTest : public ::testing::Test
+{
+protected:
+	virtual void SetUp()
+	{
+		ON_CALL(*transformer1, train_require_labels())
+		    .WillByDefault(Return(false));
+		ON_CALL(*transformer2, train_require_labels())
+		    .WillByDefault(Return(true));
+	}
+
+	Some<CPipeline> pipeline = some<CPipeline>();
+	Some<NiceMock<MockCTransformer>> transformer1 =
+	    some<NiceMock<MockCTransformer>>();
+	Some<NiceMock<MockCTransformer>> transformer2 =
+	    some<NiceMock<MockCTransformer>>();
+	Some<NiceMock<MockCMachine>> machine = some<NiceMock<MockCMachine>>();
+};
+
+TEST_F(PipelineTest, no_machine)
+{
+	pipeline->with(transformer1);
+
+	auto features = some<NiceMock<MockCFeatures>>();
+	EXPECT_THROW(pipeline->train(features), ShogunException);
```

Comment:
the need for a more specific type of exception came up in my head here atm, because expecting there ShogunException is too broad.... it could actually come from a transformer or the machine in the pipeline... right? @karlnapf @lisitsyn @vinx13